addons/product/wizard/attribute_value_wizard.py

```diff
@@ -0,0 +1,48 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+
+
+class AttributeValueWizard(models.TransientModel):
+    _name = 'attribute.value.wizard'
+    _description = 'Attribute Value Wizard'
+
+    attribute_value_id = fields.Many2one('product.attribute.value', required=True)
+    message = fields.Char(readonly=True)
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        active_id = self._context.get('default_attribute_value_id')
+        attribute_value = self.env['product.attribute.value'].browse(active_id)
+        res['attribute_value_id'] = attribute_value.id
+        if attribute_value.pav_attribute_line_ids:
+            num_products = self.env['product.template'].search_count([
+                ('attribute_line_ids.value_ids', '=', attribute_value.id)
+            ])
+            res['message'] = _(
+                "You are about to update the extra price of %s products.", num_products
+            )
+        else:
+            num_products = self.env['product.template'].search_count([
+                ('attribute_line_ids.attribute_id', '=', attribute_value.attribute_id.id)
+            ])
+            res['message'] = _(
+                "You are about to add %(attribute_value)s to %(num_products)s products.",
+                attribute_value=attribute_value.name,
+                num_products=num_products
+            )
+        return res
```

Comment:
Instead of default get IMO you should create two fields in wizard
- `Is_new_attribute_value` pass default value from action if it has attribute then false other wise true by passing `default_Is_new_attribute_value` context
- `num_products` Number of product depending on attribute_value_id field
- And message should compute base on above two field

This way it each have proper dedicated method also we don't have to rely on m2m field to get information as accessing x2m field always do query to get data

Also since it's wizard I don't we will have any issue creating fields